src/ResourceManager/Compute/Commands.Compute/Extension/AzureDiskEncryption/GetAzureDiskEncryptionStatus.cs

```diff
@@ -364,15 +415,18 @@ public override void ExecuteCmdlet()
                         WriteObject(encryptionStatus);
                         break;
                     case OSType.Linux:
-                        VirtualMachine virtualMachineResponse = this.ComputeClient.ComputeManagementClient.VirtualMachines.GetWithInstanceView(
-                            this.ResourceGroupName, VMName).Body;
-                        VirtualMachineExtension parameters = GetVmExtensionParameters(virtualMachineResponse, osType);
-
-                        this.VirtualMachineExtensionClient.CreateOrUpdateWithHttpMessagesAsync(
-                            this.ResourceGroupName,
-                            this.VMName,
-                            this.Name,
-                            parameters).GetAwaiter().GetResult();
+                        if (!IsExtensionInstalled(osType))
```

Comment:
@krkhan the function `IsExtensionInstalled` is only called when the OS type is Linux, why not Windows as well? Your function handles the logic for Windows as well.